transpiler/src/main/java/org/jsweet/transpiler/extension/RemoveJavaDependenciesAdapter.java

```diff
@@ -1526,7 +1526,10 @@ public boolean substituteNewClass(NewClassElement newClass) {
 		case "java.util.Hashtable":
 		case "java.util.WeakHashMap":
 		case "java.util.LinkedHashMap":
-			if (newClass.getArgumentCount() == 0) {
+			if (newClass.getArgumentCount() == 0 ||
+                (!types().isAssignable(newClass.getType(), newClass.getArgument(0).getType()) &&
```

Comment:
It seems that the parameter must be of type `Map<K,V>`, not necessarily of same type.
I recommend that you use the following method instead
```java
org.jsweet.transpiler.util.Util.isDeclarationOrSubClassDeclaration(...., Map.class.getName())
```

What do you think about it?

Reply:
Updated! Take a look let me know if that's what you had in mind (for example I'm not sure if there's a way to get a handle on the `ClassType`)

Reply:
Hey, this looks nice. No sadly there is not other way (AFAIK) to get a ClassType.
We could have hidden this complexity into a private method 
```java
isMap(ExtendedElement)
```
or 
```java
hasClassType(ExtendedElement, String fullClassName)
```
but it could stay as is for now :)